GV limits wrong when trim is reused

When I use a special function to adjust a GV with a trim button, then limits for the trim are applied, where limits for the GV should  be applied instead.
I have attached an OTX to test the issue. Use A up or down to adjust GVs with TrmR and TrmA, I have removed the other files, as I created a pull request.

Jesper

[Test.otx.zip](https://github.com/opentx/opentx/files/1733618/Test.otx.zip)

I managed to make a pull request to have my branch merged in, but the test build failed for AVR9X:

opentx.cpp: In function ‘uint8_t checkTrim(event_t)’:
opentx.cpp:1376:22: error: ‘GVAR_MIN’ was not declared in this scope
opentx.cpp:1376:53: error: request for member ‘min’ in ‘g_model.ModelData::gvars[((int)gvar)]’, which is of non-class type ‘gvar_t {aka int}’
opentx.cpp:1377:22: error: ‘GVAR_MAX’ was not declared in this scope
opentx.cpp:1377:53: error: request for member ‘max’ in ‘g_model.ModelData::gvars[((int)gvar)]’, which is of non-class type ‘gvar_t {aka int}’

This is all inside of a #if defined(GVARS) block, so it is pretty confusing to me. 

Do I need to fix it and make another pull request, or will someone "in the knows" be willing to take a look at it?

Thanks!
Jesper

--- a/radio/src/opentx.cpp
+++ b/radio/src/opentx.cpp
@@ -1369,35 +1369,53 @@ uint8_t checkTrim(event_t event)
       AUDIO_TRIM_MIDDLE();
       pauseEvents(event);
     }
-    else if (before>TRIM_MIN && after<=TRIM_MIN) {
-      beepTrim = true;
-      AUDIO_TRIM_MIN();
-      killEvents(event);
-    }
-    else if (before<TRIM_MAX && after>=TRIM_MAX) {
-      beepTrim = true;
-      AUDIO_TRIM_MAX();
-      killEvents(event);
-    }
-
-    if ((before<after && after>TRIM_MAX) || (before>after && after<TRIM_MIN)) {
-      if (!g_model.extendedTrims || TRIM_REUSED(idx)) after = before;
-    }
-
-    if (after < TRIM_EXTENDED_MIN) {
-      after = TRIM_EXTENDED_MIN;
-    }
-    if (after > TRIM_EXTENDED_MAX) {
-      after = TRIM_EXTENDED_MAX;
-    }
 
 #if defined(GVARS)
     if (TRIM_REUSED(idx)) {
-      SET_GVAR_VALUE(trimGvar[idx], phase, after);
+      int8_t gvar = trimGvar[idx];
+      int16_t vmin = GVAR_MIN + g_model.gvars[gvar].min;
+      int16_t vmax = GVAR_MAX - g_model.gvars[gvar].max;
+
+      if (after < vmin) {
+        after = vmin;
+        beepTrim = true;
+        AUDIO_TRIM_MIN();
+        killEvents(event);
+      }
+      else if (after > vmax) {
+        after = vmax;
+        beepTrim = true;
+        AUDIO_TRIM_MAX();
+        killEvents(event);
+      }
+
+      SET_GVAR_VALUE(gvar, phase, after);
     }
     else
 #endif
     {
+      if (before>TRIM_MIN && after<=TRIM_MIN) {
+        beepTrim = true;
+        AUDIO_TRIM_MIN();
+        killEvents(event);
+      }
+      else if (before<TRIM_MAX && after>=TRIM_MAX) {
+        beepTrim = true;
+        AUDIO_TRIM_MAX();
+        killEvents(event);
+      }
+
+      if ((before<after && after>TRIM_MAX) || (before>after && after<TRIM_MIN)) {
+        if (!g_model.extendedTrims) after = before;
+      }
+
+      if (after < TRIM_EXTENDED_MIN) {
+        after = TRIM_EXTENDED_MIN;
+      }
+      else if (after > TRIM_EXTENDED_MAX) {
+        after = TRIM_EXTENDED_MAX;
+      }
+
 #if defined(CPUARM)
       if (!setTrimValue(phase, idx, after)) {
         // we don't play a beep, so we exit now the function

--- a/radio/src/opentx.cpp
+++ b/radio/src/opentx.cpp
@@ -1373,9 +1373,14 @@ uint8_t checkTrim(event_t event)
 #if defined(GVARS)
     if (TRIM_REUSED(idx)) {
       int8_t gvar = trimGvar[idx];
+#if defined(CPUARM)
+// TODO: This was necessary to make it compile for AVR9X target - maybe there is a better solution?
       int16_t vmin = GVAR_MIN + g_model.gvars[gvar].min;
       int16_t vmax = GVAR_MAX - g_model.gvars[gvar].max;
-
+#else
+      int16_t vmin = TRIM_MIN;
+      int16_t vmax = TRIM_MAX;
+#endif
       if (after < vmin) {
         after = vmin;
         beepTrim = true;

--- a/radio/src/opentx.cpp
+++ b/radio/src/opentx.cpp
@@ -1360,6 +1360,9 @@ uint8_t checkTrim(event_t event)
     int8_t trimInc = g_model.trimInc + 1;
     int8_t v = (trimInc==-1) ? min(32, abs(before)/4+1) : (1 << trimInc); // TODO flash saving if (trimInc < 0)
     if (thro) v = 4; // if throttle trim and trim trottle then step=4
+#if defined(GVARS)
+    if (TRIM_REUSED(idx)) v = 1;
+#endif
     int16_t after = (k&1) ? before + v : before - v;   // positive = k&1
     bool beepTrim = false;
 
@@ -1374,7 +1377,6 @@ uint8_t checkTrim(event_t event)
     if (TRIM_REUSED(idx)) {
       int8_t gvar = trimGvar[idx];
 #if defined(CPUARM)
-// TODO: This was necessary to make it compile for AVR9X target - maybe there is a better solution?
       int16_t vmin = GVAR_MIN + g_model.gvars[gvar].min;
       int16_t vmax = GVAR_MAX - g_model.gvars[gvar].max;
 #else

